swaybar/ipc.c

```diff
@@ -19,13 +20,15 @@ void ipc_send_workspace_command(const char *workspace_name) {
 
 static void ipc_parse_config(struct config *config, const char *payload) {
 	json_object *bar_config = json_tokener_parse(payload);
-	json_object *tray_output, *mode, *hidden_bar, *position, *status_command;
+	json_object *tray_output, *display_mode, *hidden_state, *position, *status_command;
 	json_object *font, *bar_height, *wrap_scroll, *workspace_buttons, *strip_workspace_numbers;
 	json_object *binding_mode_indicator, *verbose, *colors, *sep_symbol, *outputs;
 	json_object *markup;
 	json_object_object_get_ex(bar_config, "tray_output", &tray_output);
-	json_object_object_get_ex(bar_config, "mode", &mode);
-	json_object_object_get_ex(bar_config, "hidden_bar", &hidden_bar);
+	// "mode" as in the bar's display mode and "mode" as in binding mode
+	// should be distinguished more carefully
+	json_object_object_get_ex(bar_config, "mode", &display_mode);
+	json_object_object_get_ex(bar_config, "hidden_state", &hidden_state);
```

Comment:
IIUC, this upgrades `hidden_state` from an internal state variable into a config option, and I'm not sure I understand why. `hidden_state` is just used a state variable for "modifier pressed" , isn't it? maybe I'm missing something.